### PR TITLE
Adding documentation on conditional attributes

### PR DESCRIPTION
**Pull request by lambda2**
#### Purpose
Adding documentation on conditional attributes for the `attribute` method.

#### Changes
Adding documentation and short example ([from this pull request](https://github.com/rails-api/active_model_serializers/pull/1403)) on conditional attributes for the `attribute` method.
#### Related GitHub issues
https://github.com/rails-api/active_model_serializers/pull/1403

### CODE REVIEW

File: docs/general/serializers.md [hunk 1]
```diff
@@ -34,6 +34,18 @@ Serialization of the resource `title`
 | `attribute :title { 'A Different Title'}` | `{ title: 'A Different Title' } `
 | `attribute :title`<br>`def title 'A Different Title' end` | `{ title: 'A Different Title' }`
 
+An `if` or `unless` parameter can make an attribute conditional. It take a symbol of a method name on the serializer.
```
Comment (beauby) [inline docs/general/serializers.md]: s/take/takes/

**Comment by beauby** [comment]
Thanks for this PR! I think we merged the PR allowing lambdas as values for the `if`/`unless` options. Would you mind adding an example using those?

File: docs/general/serializers.md [hunk 2]
```diff
@@ -34,6 +34,18 @@ Serialization of the resource `title`
...
+
+```ruby
+attribute :private_data, if: :is_current_user?
+
+def is_current_user?
+  object.id == current_user.id
+end
+```
+
 [PR please for conditional attributes:)](https://github.com/rails-api/active_model_serializers/pull/1403)
```
Comment (cgmckeever) [inline docs/general/serializers.md]: is this breadcrumb taken care of then?

Comment (beauby) [inline docs/general/serializers.md]: Yup, seems so. Good catch @cgmckeever. @lambda2 could you remove the reminder?

Comment (lambda2) [inline docs/general/serializers.md]: Yup !

**Comment by bf4** [comment]
@lambda2 great! would you like to add yourself to the changelog under misc or rebase? If not, we can handle it.

**Comment by lambda2** [comment]
@bf4 Yes, I added it under misc :)

**Comment by remear** [comment]
@lambda2 Thanks very much for this!

Could you do the following:

- Rebase to get the latest from master
- Move your CHANGELOG entry under `master (unreleased)`
- Squash your commits

Once those items are taken care of we can get this merged in.

**Comment by lambda2** [comment]
@remear yeah, of course. Seems good now !

File: docs/general/serializers.md [hunk 3]
```diff
@@ -34,7 +34,18 @@ Serialization of the resource `title`
 | `attribute :title { 'A Different Title'}` | `{ title: 'A Different Title' } `
 | `attribute :title`<br>`def title 'A Different Title' end` | `{ title: 'A Different Title' }`
 
-[PR please for conditional attributes:)](https://github.com/rails-api/active_model_serializers/pull/1403)
+An `if` or `unless` parameter can make an attribute conditional. It takes a symbol of a method name on the serializer, or a lambda literal.
```
Comment (groyoh) [inline docs/general/serializers.md]: I'd rather use `option` instead of `parameter` here.

**Comment by remear** [comment]
@lambda2, @groyoh makes a good point. Could you address his comment and squash one more time so we can wrap this one up.

File: docs/general/serializers.md (hunk 3 above)
Comment (lambda2) [inline docs/general/serializers.md]: 👌

**Comment by lambda2** [comment]
👌

**Pull request merged**